library/common/engine.cc

```diff
@@ -94,14 +95,10 @@ Engine::~Engine() {
 }
 
 void Engine::recordCounter(std::string elements, uint64_t count) {
-  if (server_) {
+  if (server_ && external_scope_) {
```

Comment:
maybe we can asset `external_scope_`? Because if the server is there but the scope is not that seems odd?

Reply:
I chose to assert both because the `server_` is still the sentinel for the dispatcher event loop, but the `Scope` could be constructed independently, and if someone changed it to be, it would silently cause a conditional on `server_` to no longer guard against the null pointer deference.

In other words, the conditional guards exactly the two pointers that are dereferenced, and it's an unenforced detail that one happens to depend on the other right now.